logprep/processor/pre_detector/processor.py

```diff
@@ -128,6 +129,8 @@ def _apply_rules(self, event, rule):
         if "@timestamp" in event:
             for detection in self._extra_data:
                 detection["@timestamp"] = event["@timestamp"]
+        for detection in self._extra_data:
+            detection["creation_timestamp"] = TimeParser.now().isoformat()
```

Comment:
How about we turn this around? That way we only have to loop over the detections once?
```suggestion
        for detection in self._extra_data:
            detection["creation_timestamp"] = TimeParser.now().isoformat()
            if "@timestamp" in event:
                detection["@timestamp"] = event["@timestamp"]
```

Reply:
Thank you for reviewing it. I have made the changes.